src/streaming/controllers/StreamController.js

```diff
@@ -801,12 +801,43 @@ function StreamController() {
      * When the playback time is updated we add the droppedFrames metric to the dash metric object
      * @private
      */
-    function _onPlaybackTimeUpdated(/*e*/) {
+    function _onPlaybackTimeUpdated(event) {
+
         if (hasVideoTrack()) {
             const playbackQuality = videoModel.getPlaybackQuality();
+
             if (playbackQuality) {
                 dashMetrics.addDroppedFrames(playbackQuality);
+
+                if(
```

Comment:
You might see this as a nitpick, but can you try to extract this into a function that is well commented vs a conditional over many lines here? It might be more readable